build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.build-logic.kotlin-dsl-gradle-plugin.gradle.kts

```diff
@@ -34,7 +34,7 @@ dependencies {
 
 tasks.withType<KotlinCompile>().configureEach {
     compilerOptions {
-        allWarningsAsErrors.set(true)
+        allWarningsAsErrors = false
```

Comment:
Let's set this back to `true`.

Reply:
Yes please :)

Reply:
Hmmm, I can't see any warning in logs. It might fail due to these two: https://github.com/gradle/gradle/blob/9a14bd5cd426bdaf22fb5ff6f0ea51649599e135/subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslCompilerPlugins.kt#L74-L77

that we suppress. 🤔

Reply:
So yeah, I left allWarningsAsErrors set to false, since compiler plugin is experimental. One option additionally is, that we don't filter the warning message. But I filter it for now, since it spams for every project.

Reply:
Could you please open an issue about moving it back to `true` when it is possible?

Reply:
Doesn't it also mean that any project where the `kotlin-dsl` plugin is applied can't use `allWarningsAsErrors`?

Reply:
Will open an issue, good idea. 

> Doesn't it also mean that any project where the kotlin-dsl plugin is applied can't use allWarningsAsErrors?

Only if they enable assignment since only then we apply a Kotlin compiler plugin. So we should at least document that wherever we will mention the assignment plugin.

Reply:
Issue: https://github.com/gradle/gradle/issues/23650